File: HLRS/openmp-test.c
#include <stdio.h>
# ifdef _OPENMP
#  include <omp.h>
# endif

int main(int argc, char** argv)
{
  int my_rank, size;

# ifdef _OPENMP
  #pragma omp parallel private(my_rank, size)
  {
    my_rank = omp_get_thread_num();
    size    = omp_get_num_threads();
    if (my_rank == 0)
    { if (size > 1)
      {
        printf ("Successful first OpenMP test executed in parallel on %i threads.\n", size);
      } else
      {
        printf ("Caution: This OpenMP test is executed only on one OpenMP thread, i.e., sequentially!\n");
      }
    }
  } //end pragma omp
# else
  printf ("Caution: Your sourcecode was compiled without switching OpenMP on\n");
  
# endif
  return 0;
}

File: HLRS/TEST/openmp-test.f90
program hello

integer :: my_rank, size
logical :: with_openmp = .FALSE.

!$ integer OMP_GET_THREAD_NUM, OMP_GET_NUM_THREADS

!$ with_openmp = .TRUE.
IF (with_openmp) THEN

!$OMP PARALLEL PRIVATE(my_rank, size)
!$ my_rank = OMP_GET_THREAD_NUM()
!$ size    = OMP_GET_NUM_THREADS()
   IF (my_rank == 0) THEN
     IF (size > 1) THEN
       WRITE(*,'(A,I3,A)') &
        & 'Successful first OpenMP test executed in parallel on ', size, ' threads.'
     ELSE
       WRITE(*,'(A)') 'Caution: This OpenMP test is executed only on one OpenMP thread, i.e., sequentially!'
     ENDIF
   ENDIF
!$OMP END PARALLEL

ELSE

  WRITE(*,'(A)') 'Caution: Your sourcecode was compiled without switching OpenMP on'

ENDIF

stop
end program hello
